core/trino-main/src/main/java/io/trino/operator/project/PageProcessor.java

```diff
@@ -342,13 +342,15 @@ private ProcessBatchResult processBatch(int batchSize)
                 }
                 else {
                     if (pageProjectWork == null) {
-                        Page inputPage = projection.getInputChannels().getInputChannels(page);
-                        expressionProfiler.start();
-                        pageProjectWork = projection.project(session, yieldSignal, inputPage, positionsBatch);
-                        long projectionTimeNanos = expressionProfiler.stop(positionsBatch.size());
-                        metrics.recordProjectionTime(projectionTimeNanos);
+                        pageProjectWork = projection.project(session, yieldSignal, projection.getInputChannels().getInputChannels(page), positionsBatch);
```

Comment:
does the same need to be done for filtering?

Reply:
filtering metrics are ok because we don't have this indirection there